File: runtime/src/main/java/com/prestongarno/ktq/Mapper.java
package com.prestongarno.ktq;

import kotlin.reflect.KProperty;

abstract class Mapper<T> {

	KProperty<?> property;

	Mapper(KProperty<?> property) {
		this.property = property;
	}

	Mapper(T provided, KProperty<?> property) {
		this.property = property;
		this.value = provided;
	}

	T value;

}

File: runtime/src/main/java/com/prestongarno/ktq/Mapper.kt
package com.prestongarno.ktq

import kotlin.reflect.KProperty

internal abstract class Mapper<T>(inst: QType, var value: T? = null, var property: KProperty<*>?) {
  constructor(inst: QType) : this(inst, null, null)

  init {
    Tracker.putProperty(inst, this, this.value)
  }

}
